src/nvim/bytes.c

```diff
@@ -0,0 +1,21 @@
+// Low-level functions with minimal dependencies, side-effects, etc.
+
+#include <assert.h>
+#include <inttypes.h>
+
+#include "nvim/vim.h"
+#include "nvim/bytes.h"
+
+#ifdef INCLUDE_GENERATED_DECLARATIONS
+# include "bytes.c.generated.h"
+#endif
+
+/// Writes time_t to "buf[8]".
+void time_to_bytes(time_t time_, uint8_t buf[8])
```

Comment:
I was planning to introduce `core.c` in a later PR to gather "stdlib wrapper functions". How about putting this there?